File: server/mc/common2/src/main/java/calebxzhou/rdi/mc/common2/chat/PlayerChatRangeState.java
package calebxzhou.rdi.mc.common2.chat;

import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

public final class PlayerChatRangeState {
    public static final ChatRange DEFAULT_RANGE = ChatRange.GLOBAL;
    private static final ConcurrentHashMap<UUID, ChatRange> CHAT_RANGES = new ConcurrentHashMap<>();

    private PlayerChatRangeState() {
    }

    public static void set(UUID playerId, ChatRange range) {
        CHAT_RANGES.put(playerId, range);
    }

    public static ChatRange get(UUID playerId) {
        return CHAT_RANGES.getOrDefault(playerId, DEFAULT_RANGE);
    }

    public static boolean isGlobal(UUID playerId) {
        return get(playerId) == ChatRange.GLOBAL;
    }

    public static void remove(UUID playerId) {
        CHAT_RANGES.remove(playerId);
    }

    public static void clear() {
        CHAT_RANGES.clear();
    }
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/chat/PlayerChatRangeState.kt
package calebxzhou.rdi.mc.rcmd.chat

import java.util.*
import java.util.concurrent.ConcurrentHashMap

object PlayerChatRangeState {
    val DEFAULT_RANGE: ChatRange = ChatRange.GLOBAL
    private val CHAT_RANGES = ConcurrentHashMap<UUID, ChatRange>()

    fun restore(playerId: UUID, store: ChatRangeStore): ChatRange {
        val range = store.load() ?: DEFAULT_RANGE
        CHAT_RANGES[playerId] = range
        return range
    }

    fun set(playerId: UUID, range: ChatRange, store: ChatRangeStore) {
        store.save(range)
        CHAT_RANGES[playerId] = range
    }

    @JvmStatic
    fun get(playerId: UUID?): ChatRange {
        return CHAT_RANGES.getOrDefault(playerId, DEFAULT_RANGE)
    }

    @JvmStatic
    fun isGlobal(playerId: UUID?): Boolean {
        return get(playerId) == ChatRange.GLOBAL
    }

    @JvmStatic
    fun remove(playerId: UUID) {
        CHAT_RANGES.remove(playerId)
    }

    @JvmStatic
    fun clear() {
        CHAT_RANGES.clear()
    }
}
